subscriptions/providers/apple_in_app/__init__.py

```diff
@@ -289,9 +337,13 @@ def _handle_refund(self, notification: AppStoreNotification) -> None:
                            transaction_info.original_transaction_id)
             refunded_payment = self._get_latest_transaction(transaction_info.original_transaction_id)
 
+        assert refunded_payment, f'Refund received for {transaction_info=} where no payments exist.'
+
         refunded_payment.subscription.end = transaction_info.revocation_date
         refunded_payment.subscription.save()
         refunded_payment.subscription_end = transaction_info.revocation_date
+        refunded_payment.status = SubscriptionPayment.Status.CANCELLED
+
```

Comment:
It's not yet established how to handle refunds, and setting as `CANCELLED` is ok, but you also need to create a `SubscriptionPaymentRefund` object.